### ISSUE TITLE
DashbuilderProcessor#buildDashbuilderWebApp throwing NullPointerException

### ISSUE THREAD

**Issue by hbelmiro** [quarkiverse/quarkus-dashbuilder#72]
I'm facing the following error when running mvn verify:

```
Error: [ERROR] 	[error]: Build step io.quarkiverse.dashbuilder.deployment.DashbuilderProcessor#buildDashbuilderWebApp threw an exception: java.lang.NullPointerException: Cannot invoke "io.quarkus.bootstrap.workspace.WorkspaceModule.getModuleDir()" because the return value of "io.quarkus.bootstrap.model.ApplicationModel.getApplicationModule()" is null
Error:  	at io.quarkiverse.dashbuilder.deployment.DashbuilderProcessor.buildDashbuilderWebApp(DashbuilderProcessor.java:157)
Error:  	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Error:  	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
Error:  	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
Error:  	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
Error:  	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
Error:  	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
Error:  	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
Error:  	at java.base/java.lang.Thread.run(Thread.java:1583)
Error:  	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

A reproducer can be found in this PR: https://github.com/hbelmiro/investing/pull/42
Build log: https://github.com/hbelmiro/investing/actions/runs/6517966647/job/17702925362?pr=42

**Comment by kjq** [comment]
Having the same issue here.  We found that building against Quarkus 3.1.2.Final works but nothing later.

**Comment by jesuino** [comment]
Hello,

Sorry the delay, I was working with other projects and didn't have time to check this.

I have faced this same NPE before, it seemed to be related to the Quarkus internal APIs when scanning resources, now it seems it is fixed with latest Quarkus versions.

The dashbuilder extension main branch is working, but I will try to make a release soon. It will have a later Dashbuilder version and all the fixes.

I will keep you posted.

**Comment by jesuino** [comment]
Hello,

I can confirm that this is fixed with the extension version 0.31.0. Please let me know if this is fixed for you and we can go ahead and close this issue.

**Comment by kjq** [comment]
Thank you.  Ill try it this week and post my update.
